### DISCUSSION TITLE
Optimizing a float parameter

### DISCUSSION THREAD

**Discussion by JasperVanhove** [kernc/backtesting.py#547] [Q&A] [answered]
Hey,

Thank you for this very usefull library. I'm looking for a way to optimize a float variable for a strategy. In my strategies i have integers where i can use the range() function but for my float variable 'atr multiplier' i can't use range. I  looked into 'arrange' and np.linspace funtions but can't seem to find a solution (neither on discussions/issues forum).

Any ideas?

**Comment by kernc** [comment, marked as answer]
Why doesn't `np.linspace` work for you?

**Reply by JasperVanhove** [reply]
How do i need to format it?
let's say i want to get use 1.0-3.0, do i right atrMultiplier=np.linspace(1.0, 3.0, num=20) ?

**Reply by kernc** [reply]
> atrMultiplier=np.linspace(1.0, 3.0, num=20)

Excatly. If that doesn't work for you, I think it's a bug in:
https://github.com/kernc/backtesting.py/blob/267d99f0d48745acf98156b2e69150c02d42f761/backtesting/backtesting.py#L1304-L1305
with `np.ndarray` not passing `isinstance` `Sequence` check.

Should work, then, if you specify it as:
```py
atrMultiplier=np.linspace(1, 3, 20).tolist()
```

**Reply by JasperVanhove** [reply]
Thank you, forget the .tolist() at the end....

**Reply by sultank1** [reply]
> > atrMultiplier=np.linspace(1.0, 3.0, num=20)
> 
> Excatly. If that doesn't work for you, I think it's a bug in:
> 
> https://github.com/kernc/backtesting.py/blob/267d99f0d48745acf98156b2e69150c02d42f761/backtesting/backtesting.py#L1304-L1305
> 
> 
> with `np.ndarray` not passing `isinstance` `Sequence` check.
> Should work, then, if you specify it as:
> 
> ```python
> atrMultiplier=np.linspace(1, 3, 20).tolist()
> ```

thanks sir

**Reply by Rydo111** [reply]
Casting `np.arange` to a list also worked for me. eg. `atrMultiplier=list(np.arange(start=0.1, stop=2.0, step=0.1))`. Thanks for this thread; helped me a lot.
